L1Trigger/L1THGCal/src/backend/HGCalShowerShape.cc

```diff
@@ -512,17 +596,70 @@ float HGCalShowerShape::sigmaRRTot(const l1t::HGCalCluster& c2d) const {
   return Srr;
 }
 
+float HGCalShowerShape::sum_X(const l1t::HGCalMulticluster& c3d, const HGCalTriggerGeometryBase& triggerGeometry, int n_layer, int offset) const {
+  const std::unordered_map<uint32_t, edm::Ptr<l1t::HGCalCluster>>& clustersPtrs = c3d.constituents();
+  unsigned nlayers =  triggerGeometry.lastTriggerLayer();
+  std::vector<double> layers(nlayers,0);
+  for (const auto& id_clu : clustersPtrs){
+    const std::unordered_map<uint32_t, edm::Ptr<l1t::HGCalTriggerCell>>& triggerCells = id_clu.second->constituents();
+  
+    for(const auto& id_tc : triggerCells) {
+      if(!pass(*id_tc.second, c3d)) continue;
+      unsigned layer = triggerGeometry.triggerLayer(id_tc.second->detId());     
+      if (layer ==0 || layer > nlayers) continue;
+      layers[layer - 1] += id_tc.second->pt(); //shift by -1 because layer 0 doesn't exist
+    } 
+  }
+  double sum_pt = 0;
+  for(int i = offset; i <= offset  + n_layer; i++)  sum_pt += layers[i];
+  double tot = 0;
+  for(unsigned i =0; i < layers.size(); ++i)  tot += layers[i];  
+  return sum_pt/tot;
+}
+
+int HGCalShowerShape::bitmap(const l1t::HGCalMulticluster& c3d,  const HGCalTriggerGeometryBase& triggerGeometry, int start, int end, float threshold) const {
+  const std::unordered_map<uint32_t, edm::Ptr<l1t::HGCalCluster>>& clustersPtrs = c3d.constituents();
+  unsigned nlayers =  triggerGeometry.lastTriggerLayer();
+  std::vector<double> layers(nlayers,0);
+  for (const auto& id_clu : clustersPtrs){
+    const std::unordered_map<uint32_t, edm::Ptr<l1t::HGCalTriggerCell>>& triggerCells = id_clu.second->constituents();
+
+    for(const auto& id_tc : triggerCells) {
+      if(!pass(*id_tc.second, c3d)) continue;
+      unsigned layer = triggerGeometry.triggerLayer(id_tc.second->detId());
+      if (layer ==0 || layer > nlayers) continue;
+      layers[layer - 1] += id_tc.second->pt(); //shift by -1 because layer 0 doesn't exist
+    }
+  }
+  //cout << "start " << start << " end " << end << " threshold " << threshold<< endl; 
+  for (float i:layers) cout << i << " ";
+  cout << endl;
+  double tot = 0;
+  for(unsigned i =0; i < layers.size(); ++i)  tot += layers[i];
+  cout << "tot pt "<< tot <<endl;
+  int bitmap = 0;
+  for(int i = start; i < end -1; i++)if(layers[i] > threshold) {
```

Comment:
The way the `for` loop is written without brace may cause issues.
To be writen like that:
```cpp
for(...) {
  if(..) {
  }
}